homeassistant/components/notify/hipchat.py

```diff
@@ -0,0 +1,105 @@
+"""
+HipChat platform for notify component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.hipchat/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    ATTR_TARGET, ATTR_DATA,
+    PLATFORM_SCHEMA, BaseNotificationService)
+from homeassistant.const import (
+    CONF_TOKEN)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['hipnotify==1.0.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_COLOR = 'color'
+CONF_ROOM = 'room'
+CONF_NOTIFY = 'notify'
+CONF_FORMAT = 'format'
+CONF_HOST = 'host'
+
+VALID_COLORS = {'yellow', 'green', 'red', 'purple', 'gray', 'random'}
+VALID_FORMATS = {'text', 'html'}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_ROOM): vol.Coerce(int),
+    vol.Optional(CONF_COLOR, default='yellow'): vol.In(VALID_COLORS),
+    vol.Optional(CONF_NOTIFY, default=False): cv.boolean,
+    vol.Optional(CONF_FORMAT, default='text'): vol.In(VALID_FORMATS),
+    vol.Optional(CONF_HOST, default='https://api.hipchat.com/'): cv.string,
+})
+
+
+def get_service(hass, config, discovery_info=None):
+    """Get the HipChat notification service."""
+    return HipchatNotificationService(
+        config[CONF_TOKEN],
+        config[CONF_ROOM],
+        config[CONF_COLOR],
+        config[CONF_NOTIFY],
+        config[CONF_FORMAT],
+        config[CONF_HOST])
+
+
+class HipchatNotificationService(BaseNotificationService):
+    """Implement the notification service for HipChat."""
+
+    def __init__(self, token, default_room, default_color, default_notify,
+                 default_format, host):
+        """Initialize the service."""
+        self._token = token
+        self._default_room = default_room
+        if default_color in VALID_COLORS:
```

Comment:
Oops, missed that when I was changing it to rely on Voluptuous to handle that (which was per a previous suggestion in the PR).